jetty-ee10/jetty-ee10-websocket/jetty-ee10-websocket-jakarta-tests/src/main/java/org/eclipse/jetty/ee10/websocket/jakarta/tests/WSServer.java

```diff
@@ -141,8 +144,20 @@ public void copyClass(Class<?> clazz) throws Exception
             assertThat("Class URL for: " + clazz, classUrl, notNullValue());
             Path destFile = classesDir.resolve(endpointPath);
             FS.ensureDirExists(destFile.getParent());
-            File srcFile = new File(classUrl.toURI());
-            IO.copy(srcFile.toPath(), destFile);
+            URI uri = classUrl.toURI();
+
+            if ("jar".equalsIgnoreCase(uri.getScheme()))
+            {
+                try (InputStream in = classUrl.openStream())
+                {
+                    Files.copy(in, destFile, StandardCopyOption.REPLACE_EXISTING);
+                }
+            }
+            else
+            {
+                File srcFile = new File(uri);
+                IO.copy(srcFile.toPath(), destFile);
+            }
```

Comment:
This feels like something our Resource class should be doing for you?

Reply:
I'm not sure why this just started failing on this PR.

But I have now changed this to use 
```java
resourceFactory.newResource(classUrl).copyTo(destFile);
```